src/darwin/Framework/CHIP/MTRDeviceController.mm

```diff
@@ -311,7 +320,53 @@ - (BOOL)isRunning
     return _cppCommissioner != nullptr;
 }
 
+- (NSUInteger)shutdownPrecheck
+{
+    __block NSUInteger assertionCount = 0;
+    dispatch_sync(_chipWorkQueue, ^{
```

Comment:
Do we actually have to do that on the work queue?  Or can we just do it under a lock?  That is, does the work queue use that value for anything?